engine/crates/engine-v2/src/engine.rs

```diff
@@ -103,6 +105,14 @@ impl<R: Runtime> Engine<R> {
             Err(response) => return HttpGraphqlResponse::build(response, format, Default::default()),
         };
 
+        if let Err(err) = self.runtime.rate_limiter().limit(&RateLimitContext::Global).await {
```

Comment:
should also be called in `execute_websocket` underneath. Maybe create_session, but would need some tweaking